File: bin/LletresRepetidesJavaScript/Fase3.js
/**
 * Fase2 - Ejercico de LletresRepetides en JavaScript
 * @author Rubén Rodríguez
 * IT Academy
 * Barcelona Activa
 *
 */
const CONSONANTS3 = "bcdfghjklmnñpqrstvwxyzBCDFGHJKLMNÑPQRSTVWXYZ";
let listNombre3 = ['R','u','b','e','n',' ','R','o','d','r','i','g','u','e','z']
let impresFs3 = false
let lletresRepetides = new Map();

var respostaFase3 = function(){
	
	let selectElement = document.getElementById('fase3Sortida')

	if (impresFs3){}
	else {
		impresFs3 = true

		listNombre3.forEach(function(letra) {
			if (lletresRepetides.has(letra)) {
				lletresRepetides.set(letra, parseInt(lletresRepetides.get(letra) )+ 1)
			}
			else 
				lletresRepetides.set(letra,1)

			if ("AEIOUaeiouáéíóúÁÉÍÓÚ".indexOf(letra) != -1) 
					selectElement.innerHTML += "Carácter "+letra+" es una VOCAL <br>"
			else if ("0123456789".indexOf(letra) != -1)
					selectElement.innerHTML += "Carácter "+letra +" ? ->Els noms de persones no contenen números! <br>"
			else if (CONSONANTS3.indexOf(letra) != -1)
					selectElement.innerHTML += "Carácter "+letra +" es una CONSONANT <br>"
			else if (letra == ' ')
					selectElement.innerHTML += "<br> "
			else
					selectElement.innerHTML += "Carácter "+letra+"  no vàlid en un nom de persona. <br>"	
		})

		lletresRepetides.forEach((value, key, self) => {
			   selectElement.innerHTML += key + ":" + value + "<br>"	

				})		
		}
				
}
	
if (document.addEventListener){
    window.addEventListener('load',(function() {
    	document.getElementById('3').onclick = respostaFase3;
    }),false);
} else {
    window.attachEvent('onload',(function() {
    	document.getElementById('3').onclick = respostaFase3;
    }));    
}	

File: src/lletreRepetidesJava/Fase3.java
/**
 * Fase3 - Ejercico de LletresRepetides
 * @author Rubén Rodríguez
 * IT Academy
 * Barcelona Activa
 *
 */

package lletreRepetidesJava;

import java.util.ArrayList;
import java.util.List;
import java.util.HashMap;

public class Fase3 {
	
	final String CONSONANTS = "bcdfghjklmnñpqrstvwxyzBCDFGHJKLMNÑPQRSTVWXYZ";
	
	private List<Character> listNombre = new ArrayList<Character>();
	HashMap<Character, Byte> repeticionsLletres = new HashMap<Character, Byte>();
	
	
	protected Fase3() {
		listNombre.add('R');
		listNombre.add('u');
		listNombre.add('b');
		listNombre.add('e');
		listNombre.add('n');
		listNombre.add(' ');
		listNombre.add('R');
		listNombre.add('o');
		listNombre.add('d');
		listNombre.add('r');
		listNombre.add('i');
		listNombre.add('g');
		listNombre.add('u');
		listNombre.add('e');
		listNombre.add('z');
	}
	
	
	protected void fase3Print () {
		
		listNombre.forEach(letra->{
			
			//Verificamos si está en el Hashmap. Si está +1 al value, sino inicialiamos.
			
			if (repeticionsLletres.containsKey(letra)) 
				repeticionsLletres.put(letra, (byte)(repeticionsLletres.get(letra)+1));
			else {
				repeticionsLletres.put(letra, (byte)1);
			}
				
			if ("AEIOUaeiouáéíóúÁÉÍÓÚ".indexOf(letra) != -1) 
		    	 System.out.println("Carácter "+letra+" es una VOCAL");
		    
			else if ("0123456789".indexOf(letra) != -1)
		    	 System.out.println("Carácter "+letra +" ? ->Els noms de persones no contenen números!");
		    
			else if (CONSONANTS.indexOf(letra) != -1)
		    	 System.out.println("Carácter "+letra +" es una CONSONANT");
		    
			else if (letra == ' ')
		    	 System.out.println(" ");
		    
			else
		    	 System.out.println("Carácter "+letra+"  no vàlid en un nom de persona.");		
			 
		});
		
		//Imprimimos por consola el hashmap completo.
		
		System.out.println("\nRepeticiones de letras del nombre");
				
		repeticionsLletres.forEach((k,v) -> {
		    System.out.println(k + "[" + v + "]");
		});
	}
	
}
	
